service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java

```diff
@@ -52,65 +47,78 @@ public void execute(CommandRequest request, Command command, Map<String, String>
         log.debug("Executing command {} for serviceId {} correlationId {} context {}", command.getCommand(), request.getServiceId(),
                 request.getCommandCorrelationId(), request.getContext());
 
+        setEnvVarsFromParameters(command, envVars);
+
         try (TerraformClient terraformClient = terraformClientFactory.createClient()) {
 
-            Path configPath = createConfigPath(request);
-            List<String> logOutput = setupTerraformClient(terraformClient, configPath);
-            String commandVerb = executeTerraformCommand(command, envVars, configPath, terraformClient);
+            Path configPath = TerraformUtils.createConfigPath(request, terraformProperties.getWorkingDirectoryRoot());
+            String commandVerb = command.getCommand();
+            List<String> logOutput = executeTerraformCommand(command, envVars, configPath, terraformClient);
             processTerraformResponse(command, commandVerb, logOutput);
         } catch (InterruptedException e) {
             log.error("Received a thread interrupt while executing the terraform command", e);
             Thread.currentThread().interrupt();
         } catch (Exception e) {
             log.error("An error was encountered while executing the terraform command", e);
-            setCommandError(command, e);
+            TerraformUtils.setCommandError(command, e);
         }
     }
 
-    private static List<String> setupTerraformClient(TerraformClient terraformClient, Path configPath) {
-        terraformClient.setWorkingDirectory(configPath.toFile());
-        List<String> output = new ArrayList<>();
-
-        // Write each terraform to the output list so that it can be processed later
-        // Also write the output to the main log to be streamed back to EP
-        terraformClient.setOutputListener(tfLog -> {
-            output.add(tfLog);
-            logToConsole(tfLog);
-        });
-        return output;
+    private static void setEnvVarsFromParameters(Command command, Map<String, String> envVars) {
+        if (command.getParameters() != null && command.getParameters().containsKey("environment") &&
+                command.getParameters().get("environment") instanceof Map) {
+            Map<String, String> environmentMap = (Map<String, String>) command.getParameters().get("environment");
+            environmentMap.keySet().forEach(key -> envVars.put("TF_VAR_" + key, environmentMap.get(key)));
```

Comment:
What env are these variables to be set in? I was thinking it was the EMA's env but it looks like it's transformed here to be exported as an environment variable later in a different context.

Reply:
The `tofu` commands are run on a process that is created by the `TerraformClient`. These environment variables need to be set on the process that is created, so `tofu` can do the appropriate variable substitution. If you look at `TerraformClient::getTerraformLauncher`, this is the method that sets up the process launcher and sets the environment variables for the process.